app/components/files/file.tsx

```diff
@@ -66,95 +77,138 @@ const File = ({
+                file={file}
+                showDate={showDate}
+                channelName={channelName}
+                onPress={handlePreviewPress}
+                theme={theme}
+            />
+        );
+    };
+
+    const renderImageFileOverlay = () => {
+        if (nonVisibleImagesCount) {
+            return (
+                <ImageFileOverlay
+                    theme={theme}
+                    value={nonVisibleImagesCount}
+                />
+            );
+        }
+        return null;
+    };
+
     const {styles, onGestureEvent, ref} = useGalleryItem(galleryIdentifier, index, handlePreviewPress);
 
-    if (isVideo(file) && publicLinkEnabled) {
+    const imageFile = () => {
         return (
             <TouchableWithoutFeedback onPress={onGestureEvent}>
-                <Animated.View style={[styles]}>
-                    <VideoFile
+                <Animated.View style={[styles, asCard ? style.imageVideo : null]}>
+                    <ImageFile
                         file={file}
                         forwardRef={ref}
                         inViewPort={inViewPort}
                         isSingleImage={isSingleImage}
                         resizeMode={'cover'}
                         wrapperWidth={wrapperWidth}
-                        updateFileForGallery={updateFileForGallery}
-                        index={index}
                     />
-                    {Boolean(nonVisibleImagesCount) &&
-                    <ImageFileOverlay
-                        theme={theme}
-                        value={nonVisibleImagesCount}
-                    />
-                    }
+                    {renderImageFileOverlay()}
                 </Animated.View>
             </TouchableWithoutFeedback>
         );
-    }
+    };
 
-    if (isImage(file)) {
+    const videoFile = () => {
         return (
             <TouchableWithoutFeedback onPress={onGestureEvent}>
-                <Animated.View style={[styles]}>
-                    <ImageFile
+                <Animated.View style={[styles, asCard ? style.imageVideo : null]}>
+                    <VideoFile
                         file={file}
                         forwardRef={ref}
                         inViewPort={inViewPort}
                         isSingleImage={isSingleImage}
                         resizeMode={'cover'}
                         wrapperWidth={wrapperWidth}
+                        updateFileForGallery={updateFileForGallery}
+                        index={index}
                     />
-                    {Boolean(nonVisibleImagesCount) &&
-                    <ImageFileOverlay
-                        theme={theme}
-                        value={nonVisibleImagesCount}
-                    />
-                    }
+                    {renderImageFileOverlay()}
                 </Animated.View>
             </TouchableWithoutFeedback>
         );
-    }
+    };
 
-    if (isDocument(file)) {
+    const documentFile = () => {
```

Comment:
0/5 nit: these functions could be better named "renderX" (i.e. renderDocumentFile)